File: epi/Ruby/13.02.epi.rb
=begin
Test for permutation of palindrome
=end
def is_permutation_of_palindrome(s)
	hash = Hash.new(0)
	s.each_char do |char|
		hash[char] += 1
	end
	puts hash.to_s
	odd_count = 0
	hash.each do |k,v|
		if (v&1) == 1
			odd_count += 1
			if odd_count > 1
				return false
			end
		end
	end
	return true
end

puts is_permutation_of_palindrome("palinlpai")

File: epi/Javascript/13.02.epi.js
/*
Test for palindromic permutations
Write a program to test whether the letters forming a string s can be permuted to form a
palindrome
*/
function isRotationOfPalindrome(s) {
    var hash = {};
    for (var i=0; i<s.length; i++) {
        if (hash[s[i]] == undefined) {
            hash[s[i]] = 0;
        }
        hash[s[i]]++;
    }
    var oddCount = 0;
    for (char in hash) {
        if (hash[char]&1) {
            oddCount++;
            if (oddCount > 2) {
                return false;
            }
        }
    }
    return true;
}
console.log(isRotationOfPalindrome("edified"));
console.log(isRotationOfPalindrome("example"));
